api/client/batch_client.py

```diff
@@ -84,11 +84,13 @@ def get_data_points(self, **selection):
         of: item_id, metric_id, region_id, frequency_id, source_id,
         partner_region_id. Additional arguments are allowed and ignored.
         """
-        headers = {'authorization': 'Bearer ' + self.access_token}
-        url = '/'.join(['https:', '', self.api_host, 'v2/data'])
-        params = lib.get_data_call_params(**selection)
-        resp = yield self.get_data(url, headers, params)
-        raise gen.Return(json_decode(resp))
+        data_points = super(BatchClient, self).get_data_points(**selection)
```

Comment:
i think we are missing a yield here?

Reply:
actually  yield would be if it was calling a coroutine. Here we are calling non-async get_data_points
https://www.tornadoweb.org/en/stable/gen.html#decorators

Reply:
doign it both ways no as discussed offline